### PR TITLE
add explanations to function's api

### PR DESCRIPTION
**Pull request by anoipm**
**Description**

Changes proposed in this pull request:

- add explanations (`kubectl explain function`) to function's api

**Related issue(s)**
See also #14998
See also #15002

### CODE REVIEW

File: components/function-controller/config/crd/bases/serverless.kyma-project.io_functions.yaml [hunk 1]
```diff
@@ -537,31 +569,47 @@ spec:
                 - minReplicas
                 type: object
               secretMounts:
+                description: SecretMounts specifies volumes containing Secrets to
+                  mount into the container's filesystem.
```
Comment (kwiatekus) [inline components/function-controller/config/crd/bases/serverless.kyma-project.io_functions.yaml:573]:
```suggestion
                description: SecretMounts specifies Secrets to
                  mount into the Function's container filesystem.
```

File: components/function-controller/config/crd/bases/serverless.kyma-project.io_functions.yaml [hunk 2]
```diff
@@ -537,31 +569,47 @@ spec:
...
                 items:
                   properties:
                     mountPath:
+                      description: MountPath specifies path within the container at
+                        which the volume should be mounted.
```
Comment (kwiatekus) [inline components/function-controller/config/crd/bases/serverless.kyma-project.io_functions.yaml:578]:
```suggestion
                        which the secret should be mounted.
```

File: components/function-controller/config/crd/bases/serverless.kyma-project.io_functions.yaml [hunk 3]
```diff
@@ -537,31 +569,47 @@ spec:
...
                       type: string
                     secretName:
+                      description: SecretName specifies name of the Secret in the
+                        Pod's Namespace to use.
```
Comment (kwiatekus) [inline components/function-controller/config/crd/bases/serverless.kyma-project.io_functions.yaml:582]:
```suggestion
                        Function's Namespace to use.
```

**Review by kwiatekus — CHANGES_REQUESTED** [review]
Looks good, ptal at the suggested changes
As far as I see those CRD schemas are repeated.. please apply in all necessary places

File: components/function-controller/config/crd/bases/serverless.kyma-project.io_functions.yaml [hunk 4]
```diff
@@ -446,17 +455,26 @@ spec:
                   type: object
                 type: array
               replicas:
+                description: Replicas defines exact number of Function's Pods to run
+                  at a time. Can't be used at the same time with ScaleConfig.
```
[the remark this answers is not in the document]
Comment (anoipm) [reply components/function-controller/config/crd/bases/serverless.kyma-project.io_functions.yaml]: fixed in .go file and regenerated

File: components/function-controller/config/crd/bases/serverless.kyma-project.io_functions.yaml [hunk 5]
```diff
@@ -518,17 +541,26 @@ spec:
...
+                description: Runtime specifies the runtime of the Function. The available
+                  values are `nodejs14`, `nodejs16`, and `python39`.
                 type: string
               runtimeImageOverride:
+                description: RuntimeImageOverride specifies the runtimes image which
+                  must be used instead of the default one.
                 type: string
               scaleConfig:
+                description: ScaleConfig defines minimum and maximum number of Function's
+                  Pods to run at a time. Can't be used at the same time with Replicas.
```
[the remark this answers is not in the document]
Comment (anoipm) [reply components/function-controller/config/crd/bases/serverless.kyma-project.io_functions.yaml]: fixed in .go file and regenerated

File: components/function-controller/config/crd/bases/serverless.kyma-project.io_functions.yaml (hunk 1 above)
Comment (anoipm) [reply components/function-controller/config/crd/bases/serverless.kyma-project.io_functions.yaml]: fixed in .go file and regenerated

File: components/function-controller/config/crd/bases/serverless.kyma-project.io_functions.yaml (hunk 2 above)
Comment (anoipm) [reply components/function-controller/config/crd/bases/serverless.kyma-project.io_functions.yaml]: fixed in .go file and regenerated

File: components/function-controller/config/crd/bases/serverless.kyma-project.io_functions.yaml [hunk 6]
```diff
@@ -518,17 +544,28 @@ spec:
...
+                  values are `nodejs14`, `nodejs16`, and `python39`.
                 type: string
               runtimeImageOverride:
+                description: RuntimeImageOverride specifies the runtimes image which
+                  must be used instead of the default one.
                 type: string
               scaleConfig:
+                description: ScaleConfig defines minimum and maximum number of Function's
+                  Pods to run at a time. When it is configured, a HorizontalPodAutoscaler
+                  will be deployed and will control the Replicas filed to scale Function
```
Comment (kwiatekus) [inline components/function-controller/config/crd/bases/serverless.kyma-project.io_functions.yaml:557]:
```suggestion
                  will be deployed and will control the Replicas field to scale Function
```

File: components/function-controller/pkg/apis/serverless/v1alpha2/function_types.go [hunk 7]
```diff
@@ -126,28 +148,40 @@ const (
...
+	// Env specifies an array of key value pairs to be used as environment variables for the Function.
+	// You can define values as static strings or reference values from ConfigMaps or Secrets.
 	Env []v1.EnvVar `json:"env,omitempty"`
 
+	// ResourceConfiguration specifies resources requested by Function and build Job.
 	// +optional
 	ResourceConfiguration *ResourceConfiguration `json:"resourceConfiguration,omitempty"`
 
+	// ScaleConfig defines minimum and maximum number of Function's Pods to run at a time.
+	// When it is configured, a HorizontalPodAutoscaler will be deployed and will control the Replicas filed
```
Comment (kwiatekus) [inline components/function-controller/pkg/apis/serverless/v1alpha2/function_types.go:170]:
```suggestion
	// When it is configured, a HorizontalPodAutoscaler will be deployed and will control the Replicas field
```

File: installation/resources/crds/serverless/crd-serverless.yaml [hunk 8]
```diff
@@ -538,17 +554,28 @@ spec:
...
+                  values are `nodejs14`, `nodejs16`, and `python39`.
                 type: string
               runtimeImageOverride:
+                description: RuntimeImageOverride specifies the runtimes image which
+                  must be used instead of the default one.
                 type: string
               scaleConfig:
+                description: ScaleConfig defines minimum and maximum number of Function's
+                  Pods to run at a time. When it is configured, a HorizontalPodAutoscaler
+                  will be deployed and will control the Replicas filed to scale Function
```
Comment (kwiatekus) [inline installation/resources/crds/serverless/crd-serverless.yaml:567]:
```suggestion
                  will be deployed and will control the Replicas field to scale Function
```

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md [hunk 9]
```diff
@@ -145,13 +145,13 @@ This table lists all the possible parameters of a given resource together with t
...
-| **spec.replicas**                             |       No       | Defines exact number of Function's Pods to run at a time. Can't be used at the same time with **spec.scaleConfig**. |
-| **spec.scaleConfig**                          |       No       | Defines minimum and maximum number of Function's Pods to run at a time. Can't be used at the same time with **spec.replicas**. |
+| **spec.replicas**                             |       No       | Defines exact number of Function's Pods to run at a time. If **spec.scaleConfig** field is configured, or if Function is targeted by external scaler, then the **spec.replicas** field is used by relevant HorizontalPodAutoscaler to control the number of active replicas. |
+| **spec.scaleConfig**                          |       No       | Defines minimum and maximum number of Function's Pods to run at a time. When it is configured, a HorizontalPodAutoscaler will be deployed and will control the **spec.replicas** filed to scale Function based on the CPU utilisation. |
```
Comment (kwiatekus) [inline docs/05-technical-reference/00-custom-resources/svls-01-function.md:149]:
```suggestion
| **spec.scaleConfig**                          |       No       | Defines minimum and maximum number of Function's Pods to run at a time. When it is configured, a HorizontalPodAutoscaler will be deployed and will control the **spec.replicas** field to scale Function based on the CPU utilisation. |
```

File: components/function-controller/config/crd/bases/serverless.kyma-project.io_functions.yaml (hunk 6 above)
Comment (anoipm) [reply components/function-controller/config/crd/bases/serverless.kyma-project.io_functions.yaml]: done

File: components/function-controller/pkg/apis/serverless/v1alpha2/function_types.go (hunk 7 above)
Comment (anoipm) [reply components/function-controller/pkg/apis/serverless/v1alpha2/function_types.go]: done

File: installation/resources/crds/serverless/crd-serverless.yaml (hunk 8 above)
Comment (anoipm) [reply installation/resources/crds/serverless/crd-serverless.yaml]: done

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md (hunk 9 above)
Comment (anoipm) [reply docs/05-technical-reference/00-custom-resources/svls-01-function.md]: done

**Review by kwiatekus — DISMISSED** [review]

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md [hunk 10]
```diff
@@ -129,7 +129,7 @@ This table lists all the possible parameters of a given resource together with t
...
 | **spec.source.gitRepository.auth.secretName** |      Yes       | Specifies the name of the Secret with credentials used by the Function Controller to authenticate to the Git repository in order to fetch the Function's source code and dependencies. This Secret must be stored in the same Namespace as the Function CR. |
-| **spec.env**                             |       No       | Specifies environment variables you need to export for the Function. You can export them either directly in the Function CR's spec or define them in a [ConfigMap](../00-configuration-parameters/svls-02-environment-variables.md#define-environment-variables-in-a-config-map). |
+| **spec.env**                             |       No       | Env specifies an array of key value pairs to be used as environment variables for the Function. You can define values as static strings or reference values from [ConfigMaps](../00-configuration-parameters/svls-02-environment-variables.md#define-environment-variables-in-a-config-map) or Secrets. |
```
Comment (grego952) [inline docs/05-technical-reference/00-custom-resources/svls-01-function.md:132]:
```suggestion
| **spec.env**                             |       No       | specifies an array of key-value pairs to be used as environment variables for the Function. You can define values as static strings or reference values from [ConfigMaps](../00-configuration-parameters/svls-02-environment-variables.md#define-environment-variables-in-a-config-map) or Secrets. |
```

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md [hunk 11]
```diff
@@ -145,13 +145,13 @@ This table lists all the possible parameters of a given resource together with t
...
 | **spec.resourceConfiguration.build.resources.requests.cpu**          |       No       | Specifies the number of CPUs requested by the build Job's Pod to operate.       |
 | **spec.resourceConfiguration.build.resources.requests.memory**       |       No       | Specifies the amount of memory requested by the build Job's Pod to operate.               |
-| **spec.replicas**                             |       No       | Defines exact number of Function's Pods to run at a time. Can't be used at the same time with **spec.scaleConfig**. |
-| **spec.scaleConfig**                          |       No       | Defines minimum and maximum number of Function's Pods to run at a time. Can't be used at the same time with **spec.replicas**. |
+| **spec.replicas**                             |       No       | Defines exact number of Function's Pods to run at a time. If **spec.scaleConfig** field is configured, or if Function is targeted by external scaler, then the **spec.replicas** field is used by relevant HorizontalPodAutoscaler to control the number of active replicas. |
```
Comment (grego952) [inline docs/05-technical-reference/00-custom-resources/svls-01-function.md:148]:
```suggestion
| **spec.replicas**                             |       No       | Defines the exact number of Function's Pods to run at a time. If **spec.scaleConfig** is configured, or if Function is targeted by an external scaler, then the **spec.replicas** field is used by the relevant HorizontalPodAutoscaler to control the number of active replicas. |
```

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md (hunk 10 above)
Comment (anoipm) [reply docs/05-technical-reference/00-custom-resources/svls-01-function.md]: fixed in .md and .go files and regenerated

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md (hunk 11 above)
Comment (anoipm) [reply docs/05-technical-reference/00-custom-resources/svls-01-function.md]: fixed in .md and .go files and regenerated

**Review by grego952 — APPROVED** [review]

**Review by kwiatekus — APPROVED** [review]

**Pull request merged**